flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -296,22 +291,11 @@ private static TaskSlotTable createTaskSlotTable(
 		final long timerServiceShutdownTimeout,
 		final int pageSize) {
 		final int numberOfSlots = (int) (1.0 / taskExecutorResourceSpec.getDefaultSlotFraction());
-		final List<ResourceProfile> resourceProfiles =
-			Collections.nCopies(numberOfSlots,
-				TaskExecutorResourceUtils.generateDefaultSlotResourceProfile(taskExecutorResourceSpec));
+		final ResourceProfile resourceProfile = TaskExecutorResourceUtils.generateDefaultSlotResourceProfile(taskExecutorResourceSpec);
 		final TimerService<AllocationID> timerService = new TimerService<>(
 			new ScheduledThreadPoolExecutor(1),
 			timerServiceShutdownTimeout);
-		return new TaskSlotTable(createTaskSlotsFromResources(resourceProfiles, pageSize), timerService);
-	}
-
-	private static List<TaskSlot> createTaskSlotsFromResources(
```

Comment:
As discussed offline, soon in later commits we no longer create the entire list of `TaskSlot` when creating `TaskSlotTable`. So there's no need to move the whole `createTaskSlotsFromResources` because it will soon be removed anyway.